zlib/grabber.go

```diff
@@ -193,6 +195,16 @@ func makeTLSConfig(config *Config, urlHost string) *ztls.Config {
 	return tlsConfig
 }
 
+func usingDefaultPort(scheme string, port uint16) bool {
+	return (scheme == "https" && port == 443) || (scheme == "http" && port == 80)
+}
+
+// Given a string of the form "host", "host:port", or "[ipv6::address]:port",
+// return true if the string includes a port, does not validate port
+func containsPort(host string) bool {
+	return strings.LastIndex(host, ":") > strings.LastIndex(host, "]")
+}
```

Comment:
I want to distinguish between the case where `SplitHostPort` fails on an absent port, in which case we keep calm and carry on, and where `SplitHostPort` fails with some other hostname parsing error, in which case we want to abort and return the error.